keps/sig-storage/3333-reconcile-default-storage-class/README.md

```diff
@@ -0,0 +1,660 @@
+
+PV controller already has all the informers it will need for this change to 
+be implemented.
+
+### Troubleshooting
+
+<!--
+This section must be completed when targeting beta to a release.
+
+For GA, this section is required: approvers should be able to confirm the
+previous answers based on experience in the field.
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+-->
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+###### What are other known failure modes?
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+## Implementation History
+
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+- 1.25: initial version
+
+## Drawbacks
+
+See "Behavior change" section above.
+
+## Alternatives
+
+### New annotation
+
+Keep the existing behavior of `storageclass.kubernetes.io/is-default-class`
+annotation as it is, but add a new SC annotation, say
+`storageclass.kubernetes.io/default-is-retroactive`, that  will retroactively
+mark all existing PVCs with `storageClassName: nil` to the new default SC.
+
+This way, we don't break existing Kubernetes behavior, trading it for a bigger
+complexity on user side - they need to use two annotations instead of one.
+
+We assume that more users will want the new behavior introduced in this KEP
+than users that want to keep the existing behavior.
+
+### Bind to `""` PVs at least once
+
+Today, if there is no default SC, a PVC with `pvc.spec.storageClassName=nil`
+will be bound to PV with `pv.spec.storageClassName=""`.
+To keep at least part of this behavior, PV controller could try to bind such
+PVCs to PV at least once and only after that change
```

Comment:
Does "only after that" here mean no candidate PV is found with `pv.spec.storageClassName=""`?

Reply:
Yes, assuming candidate PV means a PV with `pv.spec.storageClassName=""`. I've changed the sentence a bit to make it more clear.